lib/puppet/functions/stdlib/loadjson.rb

```diff
@@ -0,0 +1,84 @@
+#
+#    @summary
+#      Load a JSON file containing an array, string, or hash, and return the data
+#      in the corresponding native data type.
+#
+#      The first parameter can be a file path or a URL.
+#      The second parameter is the default value. It will be returned if the file
+#      was not found or could not be parsed.
+#
+#    @return [Array|String|Hash]
+#      The data stored in the JSON file, the type depending on the type of data that was stored.
+#
+#    @example Example Usage:
+#      $myhash = loadjson('/etc/puppet/data/myhash.json')
+#      $myhash = loadjson('https://example.local/my_hash.json')
+#      $myhash = loadjson('https://username:password@example.local/my_hash.json')
+#      $myhash = loadjson('no-file.json', {'default' => 'value'})
+#
+#
+Puppet::Functions.create_function(:'stdlib::loadjson') do
+  require 'puppet/util/json'
+
+  # If URL is in the format of https://username:password@example.local/my_info.json
+  URI_WITH_NAME_AND_PASS_PATTERN = %r{(http\://|https\://)(.*):(.*)@(.*)}
+
+  # If URL is in the format of https://username@example.local/my_info.json
+  URI_WITH_NAME_PATTERN = %r{(http\://|https\://)(.*)@(.*)}
+
+  dispatch :default_impl do
+    param 'String', :path_or_uri
```

Comment:
Could we use `Stdlib::HTTP(s)Url` here?

Reply:
I tried something in d584063 I don't know if it's correct, still need to test it. But is it more what you were thinking?